### ISSUE TITLE
Game makes strange buzzing noise when I am near zombies

### ISSUE THREAD

**Issue by Inglonias** [CleverRaven/Cataclysm-DDA#61889]
### Describe the bug

The game plays an odd buzzing noise that makes it sound like something on my computer is breaking when I am near zombies.

### Steps to reproduce

1. Ensure you are using the @ soundpack.
2. Load the attached save.
[Boreas.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/9847689/Boreas.zip)
3. Take a step in any direction to cause sounds to start playing.
4. Buzzzzzzzzzzzzzzzzzzz

### Expected behavior

No buzzing noise.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2130 (21H2)
- Game Version: cc65209 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Fewer Portal Storms [Fewer_Portal_Storms],
    Fewer Soldier Extras [Fewer_Soldiers],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    No Rail Stations [No_Rail_Stations],
    No Vault [No_Vault],
    ASCIITiles overmap color fix [colors_fix_asciitiles]
]

### Additional context

The attached file is a recording I took with my phone of the noise.
https://user-images.githubusercontent.com/2125926/197421689-df09dc0a-d72d-4bc2-8f35-1595b97111e1.mp4

**Comment by Inglonias** [comment]
You do need to take a step after loading the game for sounds to start playing

**Comment by anoobindisguise** [comment]
I was using damalsksoundpack and saw the exact same issue. When I renamed the danger_theme folder to danger_theme1, the issue went away. looks like there might be a problem with playing the danger sounds, because the actual sound effects (heartbeat) are fine in their file form.

**Comment by Inglonias** [comment]
Converting the danger_theme files in the @ soundpack from .wav to .ogg using LameXP resolves the issue for me. It looks like these files being .wav somehow causes this problem.

**Comment by anoobindisguise** [comment]
> Converting the danger_theme files in the @ soundpack from .wav to .ogg using LameXP resolves the issue for me. It looks like these files being .wav somehow causes this problem.

I moved over some dummy oggs and changed the effects file to reference them and things were working correctly too

**Comment by onura46** [comment]
I can confirm this happening in this save and in my own games. Fwiw, I tested whether stripping the ID3 tags from the WAV files would help (sometimes programs don't expect the extra metadata), and it did not help. Transcoding to .ogg did!

**Comment by akrieger** [comment]
So we did recently rev the vcpkg commit because zlib took down a release we depended on. But nothing changed in sdl2-mixer (or any sdl), a dependency libflac updated to 1.3.4, but that's irrelevant cause WAV is natively supported by sdl. So this is puzzling.

**Comment by akrieger** [comment]
Unfortunately I can't repro the issue even with your save & the @ soundfile I downloaded from like, 2019, after forcing a rebuild with the newer vcpkg version.

**Comment by Inglonias** [comment]
Alright, so this isnt a universal issue, which means I need to share my hardware specs.

The computer I saw this issue on was this model of Acer Aspire 5: https://laptopmedia.com/laptop-specs/acer-aspire-5-a515-54g-5928/

**Comment by Inglonias** [comment]
This is not the same computer I initially reported the issue on, but the buzzing occurs here too.
![image](https://user-images.githubusercontent.com/2125926/197824198-e57f1b75-5e93-4420-88fd-9f785403eeb5.png)

I will check the other flavors of build and report back with the results

**Comment by Inglonias** [comment]
It looks like it's just the non-msvc 64-bit build that's buzzing on my desktop computer. I tried four different flavors of Cataclysm and noted what happened when I got near the zombies. The build hash I used for testing was 577ab49

64-bit: Buzz
64-bit msvc: No buzz
32-bit: No buzz.
32-bit msvc: No buzz

**Comment by Inglonias** [comment]
Tested and took correct notes this time.

64-bit: No buzz.
**64-bit msvc: Buzz.**
32-bit: No buzz.
32-bit msvc: No buzz

**Comment by akrieger** [comment]
I managed to reproduce audio issues with the 32 and 64 bit msvc cdda build from github. Not exactly as described here, it's like a few tenths of a second of static every few seconds in 32 bit and a squeaking in 64 bit, but it's something. It wasn't happening with anything I built myself locally. I'll continue to try various things to see why it might be happening.

**Comment by akrieger** [comment]
Github builds reproduce the issue with `windows-latest` runner which is using VS2022, so it ain't that.

**Comment by akrieger** [comment]
Reproduces with the most recent vcpkg commit including zlib 1.12.3 update. I'm going to setup a custom vcpkg repo with the old 'known good' commit and cherry pick patches to it until it builds and see if I can bisect that way.

**Comment by akrieger** [comment]
*rubs eyes* so I've managed to still reproduce audio issues with builds using the old vcpkg commit hacked up so it's building against the old zlib still, which means one or more of:

1) something changed in the runners which is causing the breakage, like the statically linked crt is broken somehow.
2) something changed in cdda code which caused this breakage.
3) this has always been broken and we're only now noticing.

I'm going to try downloading older releases of cdda to see if it is happening there.

**Comment by akrieger** [comment]
Does not repro on https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-10-19-0515, so that gives us a bisect start point.

**Comment by akrieger** [comment]
I don't really understand why, but https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-10-20-0439 was the last release which doesn't reproduce this for me, and https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-10-20-1349 is the first which does. Can anyone here check that this matches their experience?

**Comment by akrieger** [comment]
Looking at the build logs, the builds which don't reproduce this show they're using sdl2-mixer v2.0.4 which matches linux.
```
             sdl2-mixer[core,libflac,libmodplug,libvorbis,mpg123,nativemidi]:x64-windows-static -> 2.0.4#16
```

The builds which do reproduce this show they're using sdl2-mixer x2.6.1
```
             sdl2-mixer[core,libflac,libmodplug,libvorbis,mpg123,nativemidi]:x64-windows-static -> 2.6.1#2
```

Currently trying to track down why that is because AFAICT the current version of sdl2-mixer in vcpkg is still 2.0.4.

**Comment by akrieger** [comment]
So something changed in the runner image and the vcpkg setup by the `run-vcpkg` step is no longer the vcpkg being used in the build.

Good builds:
```
         "D:\a\Cataclysm-DDA\b\vcpkg\vcpkg.exe" install "--host-triplet=x64-windows-static" --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "D:\a\Cataclysm-DDA\b\vcpkg\\" "--x-manifest-root=D:\a\Cataclysm-DDA\Cataclysm-DDA\msvc-full-features\\" "--x-install-root=D:\a\Cataclysm-DDA\Cataclysm-DDA\msvc-full-features\vcpkg_installed\x64-windows-static\\" --clean-after-build
```
Bad builds:
```
"C:\vcpkg\vcpkg.exe" install "--host-triplet=x64-windows-static" --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "C:\vcpkg\\" "--x-manifest-root=D:\a\Cataclysm-DDA\Cataclysm-DDA\msvc-full-features\\" "--x-install-root=D:\a\Cataclysm-DDA\Cataclysm-DDA\msvc-full-features\vcpkg_installed\x64-windows-static\\" --clean-after-build
```

**Comment by akrieger** [comment]
Ok, I tracked down why this is happening. The run-vcpkg step's vcpkg.exe doesn't seem to be getting used anymore, resulting in cdda building against a fully up to date master vcpkg installed on the system. This ends up pulling in the updated sdl2-mixer package which has the bug. I filed an issue on `run-vcpkg` for awareness https://github.com/lukka/run-vcpkg/issues/170. https://github.com/libsdl-org/SDL_mixer/issues/466 seems to be the relevant issue on the mixer side for the regression.

I'm going to fix the builds by properly forcing the right vcpkg.exe to be used.
